src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -25,538 +59,574 @@
-            Integer userId = user.getUserId();
-            List<Map<String, String>> datasets = datasetService.autoCompleteDatasets(partial, userId);
-            return Response.ok(datasets, MediaType.APPLICATION_JSON).build();
-        } catch (Exception e) {
-            return createExceptionResponse(e);
-        }
+  }
+
+  @GET
+  @Path("/autocomplete/{partial}")
+  @Produces("application/json")
+  @PermitAll
+  @Deprecated
+  public Response datasetAutocomplete(@Auth AuthUser authUser,
+      @PathParam("partial") String partial) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      Integer userId = user.getUserId();
+      List<Map<String, String>> datasets = datasetService.autoCompleteDatasets(partial, userId);
+      return Response.ok(datasets, MediaType.APPLICATION_JSON).build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
     }
-
-    @GET
-    @Path("/search")
-    @Produces("application/json")
-    @PermitAll
-    public Response searchDatasets(@Auth AuthUser authUser, @QueryParam("query") String query) {
-        try {
-            User user = userService.findUserByEmail(authUser.getEmail());
-            List<Dataset> datasets = datasetService.searchDatasets(query, user);
-            return Response.ok().entity(unmarshal(datasets)).build();
-        } catch (Exception e) {
-            return createExceptionResponse(e);
-        }
+  }
+
+  @GET
+  @Path("/search")
+  @Produces("application/json")
+  @PermitAll
+  public Response searchDatasets(@Auth AuthUser authUser, @QueryParam("query") String query) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      List<Dataset> datasets = datasetService.searchDatasets(query, user);
+      return Response.ok().entity(unmarshal(datasets)).build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
     }
-
-    @PUT
-    @Produces("application/json")
-    @RolesAllowed(ADMIN)
-    public Response updateNeedsReviewDataSets(@QueryParam("dataSetId") Integer dataSetId, @QueryParam("needsApproval") Boolean needsApproval) {
-        try {
-            Dataset dataset = datasetService.updateNeedsReviewDatasets(dataSetId, needsApproval);
-            return Response.ok().entity(unmarshal(dataset)).build();
-        } catch (Exception e) {
-            return createExceptionResponse(e);
-        }
+  }
+
+  @POST
+  @Path("/search/index")
+  @Consumes("application/json")
+  @Produces("application/json")
+  public Response searchDatasetIndex(@Auth AuthUser authUser, String query) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      // Use the searchDataset method as a dummy method to validate the query
+      List<Dataset> datasets = datasetService.searchDatasets(query, user);
+      return Response.ok().entity(unmarshal(datasets)).build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
     }
```

Comment:
Here's what was added.

Reply:
For this ticket, I think we need a little more to facilitate UI development. The return object should be a list of terms as defined in https://broadworkbench.atlassian.net/browse/DUOS-2469 ... it doesn't have to be exact, but should be something that has that structure so the UI devs can manipulate those terms in components.

Also, the input object should also be a json structure that looks like a term as defined in DUOS-2469. That way the UI can make real calls with a term query object and theoretically get back a filtered result. 

Lastly, this method should provide a minimal amount of filtering on the mock terms. If we had, say, 5 terms with some pre-defined filter fields, that should be enough for UI work.